server/src-lib/Hasura/Db.hs

```diff
@@ -56,28 +58,28 @@ instance (MonadTx m) => MonadTx (ValidateT e m) where
 data LazyTx e a
   = LTErr !e
   | LTNoTx !a
-  | LTTx !(Q.TxE e a)
+  | LTTx !(Q.TxE e a) !(Maybe Q.TxAccess)
 
 lazyTxToQTx :: LazyTx e a -> Q.TxE e a
 lazyTxToQTx = \case
   LTErr e  -> throwError e
   LTNoTx r -> return r
-  LTTx tx  -> tx
+  LTTx tx _txaM -> tx
 
 runLazyTx
   :: PGExecCtx
   -> LazyTx QErr a -> ExceptT QErr IO a
 runLazyTx (PGExecCtx pgPool txIso) = \case
   LTErr e  -> throwError e
   LTNoTx a -> return a
-  LTTx tx  -> Q.runTx pgPool (txIso, Nothing) tx
+  LTTx tx txaM -> Q.runTx pgPool (txIso, txaM) tx
 
 runLazyTx'
   :: PGExecCtx -> LazyTx QErr a -> ExceptT QErr IO a
-runLazyTx' (PGExecCtx pgPool _) = \case
+runLazyTx' (PGExecCtx pgPool txIso) = \case
   LTErr e  -> throwError e
   LTNoTx a -> return a
-  LTTx tx  -> Q.runTx' pgPool tx
+  LTTx tx txaM  -> Q.runTx pgPool (txIso, txaM) tx
```

Comment:
We should be using `Q.runTx'` here?
`runLazyTx'` are non-transactional queries (Without `BEGIN`/`COMMIT`/`ABORT`). It's being used to run subscription queries.

Reply:
Ah interesting. Another break.